lib/ruby_lsp/requests/document_link.rb

```diff
@@ -63,6 +72,53 @@ def gem_paths
             lookup
           end, T.nilable(T::Hash[String, T::Array[String]]))
         end
+
+        sig { void }
+        def fetch_rails_documents
+          @rails_documents = T.let(begin
```

Comment:
To get around the nilable thing, you can probably rename this method to `rails_documents` and use conditional assignment, which Sorbet now supports. Something like
```ruby
sig { returns(T::Hash[String, String]) }
def rails_documents
  @rails_documents ||= T.let(..., T.nilable(T::Hash[String, String]))
end
```
We know we'll always set it to a hash in fail cases, so it should only execute once.